launch_ros/launch_ros/descriptions/composable_node.py

```diff
@@ -76,6 +78,42 @@ def __init__(
         if extra_arguments:
             self.__extra_arguments = normalize_parameters(extra_arguments)
 
+    @classmethod
+    def parse(cls, parser: Parser, entity: Entity):
+        """Parse composable_node."""
+        from launch_ros.actions import Node
+        kwargs = {}
+
+        kwargs['package'] = parser.parse_substitution(entity.get_attr('pkg'))
+        kwargs['plugin'] = parser.parse_substitution(entity.get_attr('plugin'))
+        kwargs['name'] = parser.parse_substitution(entity.get_attr('name'))
+
+        namespace = entity.get_attr('namespace', optional=True)
+        if namespace is not None:
+            kwargs['namespace'] = parser.parse_substitution(namespace)
+
+        parameters = entity.get_attr('param', data_type=List[Entity], optional=True)
+        if parameters is not None:
+            kwargs['parameters'] = Node.parse_nested_parameters(parameters, parser)
+
+        remappings = entity.get_attr('remap', data_type=List[Entity], optional=True)
+        if remappings is not None:
+            kwargs['remappings'] = [
+                (
+                    parser.parse_substitution(remap.get_attr('from')),
+                    parser.parse_substitution(remap.get_attr('to'))
+                ) for remap in remappings
+            ]
+
+            for remap in remappings:
+                remap.assert_entity_completely_parsed()
+
+        extra_arguments = entity.get_attr('extra_arg', data_type=List[Entity], optional=True)
+        if extra_arguments is not None:
+            kwargs['extra_arguments'] = Node.parse_nested_parameters(extra_arguments, parser)
```

Comment:
I don't think that reusing `parse_nested_parameters` here works.
I'm not sure why `extra_arguments` was originally documented as `SomeParameters`, is that okay @hidmic ?

Reply:
Okay.

It doesn't seem that reusing `Node.parse_nested_parameters` will work here, as that will allow:

```xml
<extra_arg>
  <param>name="asd" value="bsd"</param>
</extra_arg>
```

or

```xml
<extra_arg name="asd" value="bsd">
```

The first example (nested param), doesn't make much sense for `extra_arg`.
I think that the same applies to the constructor, a nested dict doesn't make much sense.

Reply:
I agree that `<param/>` tag is confusing. But the nested dictionary is what the action currently allows.

As I said above, these bits probably need a revisit. To not block this PR, @kenji-miyake, could you parse `<extra_arg/>` tags manually and put together a dictionary of extra arguments instead of delegating to `Node.parse_nested_parameters`? Meanwhile, we have to open a ticket to follow-up with this.

Reply:
@hidmic Okay, I'll try it.
But actually, I don't understand what is `SomeParameters` and what `Node.parse_nested_parameters` is doing...

Also, by the way, is there any way to check whether `extra_arg` is properly set to a node?
I think it can't be seen by `ros2 param list`.

Reply:
you need to pass the complete substitution not only the `0` entry:

```python3
extra_arguments = entity.get_attr('extra_arg', data_type=List[Entity], optional=True)
if extra_arguments is not None:
    kwargs['extra_arguments'] = [
        {
            tuple(parser.parse_substitution(extra_arg.get_attr('name'))):
            parser.parse_substitution(extra_arg.get_attr('value'))
        } for extra_arg in extra_arguments
    ]
    for extra_arg in extra_arguments:
        extra_arg.assert_entity_completely_parsed()
```

(the cast from `list` to `tuple` is because `list` isn't hashable)